dragonfly/room2d.py

```diff
@@ -1063,40 +1075,36 @@ def _honeybee_plenums(self, hb_room, tolerance=0.01):
                     If there isn't enough space between the Story floor_height and
                     the Room2D floor_height, this item will be None.
         """
-
-        hb_rooms = [None, None]
-
-        try:
-            parent = self.parent
-        except AttributeError:
-            print('Cannot add plenums to the {} Room because the parent Story has '
-                  'not been set. This is required to derive the plenum '
-                  'height.'.format(self.identifier))
-            raise
-
-        ceil_plenum_height = \
-            parent.floor_to_floor_height - self.floor_to_ceiling_height
+        # check to be sure that the room2d has a parent story
+        hb_rooms = []
+        if not self.has_parent:
+            raise AttributeError(
+                'Cannot add plenums to the "{}" Room because the parent Story has '
+                'not been set. This is required to derive the plenum '
+                'height.'.format(self.identifier))
+
+        parent = self.parent
+        parent_ceiling = parent.floor_height + parent.floor_to_floor_height
+        ceil_plenum_height = parent_ceiling - self.ceiling_height
```

Comment:
@chriswmackey 

I might be missing something here, isn't this calculating the same value as:
`ceil_plenum_height = parent.floor_to_floor_height - self.floor_to_ceiling_height` ?

Reply:
No. It's subtle but it's different. In the current case, we derive the ceiling plenum height from the difference between the story ceiling height and the room ceiling height. Before, you were just subtracting the floor-to-ceiling height from the floor-to-floor height, which will not work in the case of floor plenums.